Library/Homebrew/rubocops/shared/desc_helper.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require "rubocops/shared/string_content"
+require "rubocops/shared/helper_functions"
+
+# TODO: move the micro shared modules above into this one?
+
+module RuboCop
+  module Cop
+    module DescHelper
+      include StringContent
+      include HelperFunctions
+
+      VALID_LOWERCASE_WORDS = %w[
+        macOS
+      ].freeze
+
+      def audit_desc(type, name, desc_call)
+        # Check if a desc is present
+        if desc_call.nil?
+          problem "#{type.to_s.capitalize} should have a desc (Description)."
+          return
+        end
+
+        desc = desc_call.first_argument
+
+        # Check the desc length. Should be >0 and <80 characters.
+        pure_desc_length = string_content(desc).length
+        if pure_desc_length.zero?
+          problem "The desc (description) should not be an empty string."
+          return
+        end
+
+        # Check for leading whitespace.
+        problem "Description shouldn't have a leading space" if regex_match_group(desc, /^\s+/)
+
+        # Check for trailing whitespace.
+        problem "Description shouldn't have a trailing space" if regex_match_group(desc, /\s+$/)
+
+        # Check if command-line is wrongly used in the desc
```

Comment:
Just a nit, but some of these comments end in a period, others don't. What should be the standard?

Reply:
All should end with a period I'd say.

Reply:
I think some of these comments could also be removed if the `problem` string is self-explanatory.

Reply:
I added the periods and reworded some of the comments for clarity and consistency. I elected to keep them as visual delimiters between each check, since the code structure differs a bit between the checks.